sql/core/src/main/scala/org/apache/spark/sql/execution/ExternalAppendOnlyUnsafeRowArray.scala

```diff
@@ -228,6 +234,11 @@ private[sql] class ExternalAppendOnlyUnsafeRowArray(
       currentRow.pointTo(iterator.getBaseObject, iterator.getBaseOffset, iterator.getRecordLength)
       currentRow
     }
+
+    override protected def closeIfNeed(): Unit = iterator match {
```

Comment:
IfNeed -> IfNeeded

Reply:
[c8f6f08](https://github.com/apache/spark/pull/36262/commits/c8f6f0871f5875433d5dc951805a7e40d875b7ee) fix this